app/bundles/PluginBundle/Bundle/PluginBundleBase.php

```diff
@@ -33,28 +33,44 @@ public static function onPluginInstall(Plugin $plugin, MauticFactory $factory, $
     public static function installPluginSchema(array $metadata, MauticFactory $factory, $installedSchema = null): void
     {
         if (null !== $installedSchema) {
-            // Schema exists so bail
+            // Schema already exists, so no need to proceed
             return;
         }
 
         $db             = $factory->getDatabase();
         $schemaTool     = new SchemaTool($factory->getEntityManager());
         $installQueries = $schemaTool->getCreateSchemaSql($metadata);
 
-        $db->beginTransaction();
-        try {
-            foreach ($installQueries as $q) {
-                $db->executeQuery($q);
+        foreach ($installQueries as $q) {
+            try {
+                // Check if the query is a DDL statement
+                if (self::isDDLStatement($q)) {
+                    // Execute DDL statements outside of a transaction
+                    $db->executeQuery($q);
+                } else {
+                    // For non-DDL statements, use transactions
+                    try {
+                        $db->beginTransaction();
+                        $db->executeQuery($q);
+                        $db->commit();
+                    } catch (\Exception $e) {
```

Comment:
Please catch the specific exception you want to catch. I saw many times that it's masking other exceptions and it makes debugging very time consuming. For example it will mute the deprecation notices. In the tests it will catch the wrong assertion exceptions. We should refactor all the "catch all" occurances.

Reply:
I removed the catch exception at all. I just let catch exception If we can make rollback, but anyway, it continues in exception.